src/annotations.jl

```diff
@@ -0,0 +1,154 @@
+
+Return the `*.onda.annotations.arrow`-compliant table read from `io_or_path`.
+
+If `validate_schema` is `true`, the table's schema will be validated to ensure it is
+a `*.onda.annotations.arrow`-compliant table. An `ArgumentError` will be thrown if
+any schema violation is detected.
+
+If `materialize` is `false`, the returned table will be an `Arrow.Table` while if
+`materialize` is `true`, the returned table will be a `NamedTuple` of columns. The
+primary difference is that the former has a conversion-on-access behavior, while
+for the latter, any potential conversion cost has been paid up front.
+"""
+function read_annotations(io_or_path; materialize::Bool=false, validate_schema::Bool=true)
+    table = read_onda_table(io_or_path; materialize)
+    validate_schema && validate_annotation_schema(Tables.schema(table))
+    return table
+end
+
+"""
+    write_annotations(io_or_path, table; kwargs...)
+
+Write `table` to `io_or_path`, first validating that `table` is a
+`*.onda.annotations.arrow`-compliant table. An `ArgumentError` will
+be thrown if any schema violation is detected.
+
+`kwargs` is forwarded to an internal invocation of `Arrow.write(...; file=true, kwargs...)`.
+"""
+function write_annotations(io_or_path, table; kwargs...)
+    columns = Tables.columns(table)
+    schema = Tables.schema(columns)
+    try
+        validate_annotation_schema(schema)
+    catch
+        @warn "Invalid schema in input `table`. Try calling `Annotation.(Tables.rows(table))` to see if it is convertible to the required schema."
+        rethrow()
+    end
+    return write_onda_table(io_or_path, columns; kwargs...)
+end
+
+#####
+##### utilities
+#####
+
+"""
+    merge_overlapping_annotations(annotations)
+
+Given the `*.onda.annotations.arrow`-compliant table `annotations`, return
+a table corresponding to `annotations` except that overlapping entries have
+been merged.
+
+Specifically, two annotations `a` and `b` are determined to be "overlapping"
+if `a.recording == b.recording && TimeSpans.overlaps(a.span, b.span)`. Merged
+annotations' `span` fields are generated via calling `TimeSpans.shortest_timespan_containing`
+on the overlapping set of source annotations.
+
+The returned annotations table only has a single custom column named `from`
+whose entries are `Vector{UUID}`s populated with the `id`s of the generated
+annotations' source(s). Note that every annotation in the returned table
+has a freshly generated `id` field and a non-empty `from` field, even if
+the `from` only has a single element (i.e. corresponds to a single
+non-overlapping annotation).
+
+Note that this function internally works with `Tables.columns(annotations)`
+rather than `annotations` directly, so it may be slower and/or require more
+memory if `!Tables.columnaccess(annotations)`.
+"""
+function merge_overlapping_annotations(annotations)
+    columns = Tables.columns(annotations)
+    merged = Annotation[]
+    for (rid, (locs,)) in Onda.locations((columns.recording,))
```

Comment:
```suggestion
    for (rid, (locs,)) in locations((columns.recording,))
```
You don't need the module qualification inside of that module.